magento/magento2
# Issue-33519. Added logic for grouping alerts by store

### Description (*)
Added logic for grouping alerts by store. The problem was that when sorting collection items, the store ID of the last item of current user is always add to the email object. This fix also fixed the situation with PriceAlerts, because the logic is used by everyone.

### Fixed Issues (if relevant)
1. Fixes magento/magento2#33519

### Manual testing scenarios (*)
1. Setup Multiple Stores/websites in Magento
2. Create 2 different Stock Alert Email Templates for each store (Marketing - Communications - Email Templates - Add New Template)
3. Enable Product Alerts and Apply the newly created templates for each Store
4. Stores - Configuration - Catalog - Catalog - Product Alerts - Stock Alert Email Template - Set the new template created for new Store view scope , "Allow Alert When Product Comes Back in Stock" - Yes
Default Scope: Product Alerts Run Settings - Give the Start time and Save Configuration
Change the Scope of Store View to another Store view and apply other newly created template and save the configuration
5. Reindex and clear cache
6. Admin - Add a new Product - Set the Product to Out Of Stock and Save
7. Front-end - New Store website (for ex: France website) - Sign in as customer - Product Details page - Click on "Notify me when this product is in stock"
8. Open the same Product in different Store (for ex: Germany website) - Sign in as same customer - Product Details page - Click on "Notify me when this product is in stock"
9. Admin - Edit the same product and Set the stock status to "In Stock" and Save
10. Run the Cron > php bin/magento cron:run
11. Verify emails of both the Stores(You should receive two emails from both Stores with different email templates)

### Contribution checklist (*)
